h2o-core/src/main/java/water/rapids/Merge.java

```diff
@@ -52,6 +52,15 @@ public static Frame sort(final Frame fr, int[] cols, int[] ascending) {
   public static Frame merge(final Frame leftFrame, final Frame riteFrame, final int leftCols[], final int riteCols[],
                             boolean allLeft, int[][] id_maps) {
     int[] ascendingL, ascendingR;
+    
+    if (leftFrame.numRows()==0 && riteFrame.numRows() > 0) { // if one frame contains zero rows, returns the other frame which may contain zero rows
+      return riteFrame;
+    }
+    if (riteFrame.numRows()==0 && leftFrame.numRows() > 0) {
+      return leftFrame;
+    }
+    if (riteFrame.numRows()==0 && leftFrame.numRows()==0)
+      throw new IllegalArgumentException("Cannot perform merge on empty frames.");
```

Comment:
why not?

Reply:
That should not be hard to do to just return an empty frame.

Reply:
Honza:

If both frames are empty, the original code will return an frame with zero rows.  Hence, I remove my change and it works as you requested.